relay-config/src/config.rs

```diff
@@ -2282,7 +2282,8 @@ impl Config {
             max_connections: redis
                 .options
                 .max_connections
-                .unwrap_or(cpu_concurrency as u32 * 2),
+                .unwrap_or(cpu_concurrency as u32 * 2)
+                .min(24),
```

Comment:
Can we make this a documented constant? Even if there's no strong reason to pick exactly this number, that's also useful information to future readers, e.g.

```rust
/// For small setups, `2 x limits.max_thread_count` does not leave enough headroom.
/// In this case, we fall back to the old default.
const DEFAULT_MAX_REDIS_CONNECTIONS: u32 = 24;
```

```suggestion
                .min(DEFAULT_MAX_REDIS_CONNECTIONS),
```